### ISSUE TITLE
ThrottlingProtocol incorrectly calls stopReading and startReading on its transport

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11210]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1728|
|Type|defect|
|Created|2006-05-19 18:32:15Z|

It means to be calling pauseProducing and resumeProducing.

<details><summary>Searchable metadata</summary>

```
trac-id__1728 1728
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1148063535000000 1148063535000000
changetime__1148064748000000 1148064748000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Here is a fix:

```
Index: twisted/protocols/policies.py
===================================================================
--- twisted/protocols/policies.py       (revision 16870)
+++ twisted/protocols/policies.py       (working copy)
@@ -141,10 +141,10 @@
 
 
     def throttleReads(self):
-        self.transport.stopReading()
+        self.transport.pauseProducing()
 
     def unthrottleReads(self):
-        self.transport.startReading()
+        self.transport.resumeProducing()
 
     def throttleWrites(self):
         if hasattr(self, "producer"):
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @radix|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @exarkun|
|-|-|

It is a +1. Man, why do people gotta use non-interfaces.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16878]) Apply [#11210](https://github.com/twisted/twisted/issues/11210) patch - call methods of public, document interface to control traffic flow

Author: exarkun
Reviewer: radix
Fixes [#11210](https://github.com/twisted/twisted/issues/11210)

pauseProducing and resumeProducing are methods of the IProducer interface, which
a protocol's transport actually provides.  (stop|start)(Reading|Writing) are
private implementation details which shouldn't be called outside of reactor
implementation code.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
